From 491dbfe072a6ac15da28c581eba2b885cf235051 Mon Sep 17 00:00:00 2001
From: Tobias Geerinckx <tobias.geerinckx@gmail.com>
Date: Tue, 3 Jun 2014 04:59:34 +0200
Subject: [PATCH] lsinitrd.sh: don't choke on LZO-compressed images

lsinitrd (and hence dracut) currently fail to list the contents of any
LZO-compressed image, and merely spit out misleading xzcat errors.

I guess no-one actually uses them.

(cherry picked from commit 773d6a7ded0cc717ecfb7bed6e3f52ceedfb1834)
---
 lsinitrd.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lsinitrd.sh b/lsinitrd.sh
index 7b12ab96..b36b34e2 100755
--- a/lsinitrd.sh
+++ b/lsinitrd.sh
@@ -181,6 +181,9 @@ CAT=$({
             $'\x02\x21'*)
                 echo "lz4 -d -c"
                 ;;
+            $'\x89'LZO$'\0'*)
+                echo "lzop -d -c"
+                ;;
             *)
                 if echo "test"|xz|xzcat --single-stream >/dev/null 2>&1; then
                     echo "xzcat --single-stream --"
